src/qt/walletcontroller.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_QT_WALLETCONTROLLER_H
+#define BITCOIN_QT_WALLETCONTROLLER_H
+
+#include <qt/walletmodel.h>
+
+#include <list>
+#include <memory>
+#include <vector>
+
+class BitcoinGUI;
+class OptionsModel;
+class PlatformStyle;
+
+namespace interfaces {
+class Handler;
+class Node;
+} // namespace interfaces
+
+/**
+ * WalletController
+ */
+class WalletController : public QObject
+{
+    Q_OBJECT
+
+public:
+    WalletController(interfaces::Node& node, const PlatformStyle* platform_style, OptionsModel* options_model, QObject* parent);
+    ~WalletController();
+
+    void setWindow(BitcoinGUI* window);
+
+public Q_SLOTS:
+    void setCurrentWallet(WalletModel* wallet_model);
+
+private Q_SLOTS:
+    void addWallet(WalletModel* wallet_model);
+
+Q_SIGNALS:
+    void walletAdded(WalletModel* wallet_model);
+    void walletRemoved(WalletModel* wallet_model);
+    void currentWalletChanged(WalletModel* wallet_model);
+
+    void coinsSent(WalletModel* wallet_model, SendCoinsRecipient recipient, QByteArray transaction);
+
+private:
+    interfaces::Node& m_node;
```

Comment:
do `m_node`, `m_platform_style`, `m_options_model` and `m_handler_load_wallet` need to be member variables?

Reply:
Yes, to create `WalletModel` instances. And `m_handler_load_wallet` is needed to keep the signal/slot connection.